backend/app/routes/admin/updateAdmin.js

```diff
@@ -0,0 +1,36 @@
+const to = require('await-to-js').default;
+const Admin = require('../../models/Admin');
+const { ErrorHandler } = require('../../../helpers/error');
+const constants = require('../../../constants');
+
+module.exports = async (req, res, next) => {
+  const userRecord = await Admin.findOne({
+    email: res.locals.decode.email,
+  });
+
+  // Finding user through email which needs to be updated
+  const user = await Admin.findOne({ email: userRecord.email });
+
+  // Checking which fields are present and making changes to the present field
+  if (user) {
+    user.username = req.body.username || user.username;
```

Comment:
What if the new username is already in use by someone else? How will your code handle that or share the same information with the frontend